### ISSUE TITLE
A11y Issue: Keyboard/Tab Trap

### ISSUE THREAD

**Issue by qkevinto** [jaredreich/pell#66]
Keyboard trapping occurs due to the tab key feature (#13). When a keyboard user tabs through an the interface and reaches the Pell editor, they will be stuck inside it if they are without mouse control. There's also the issue of missing WAI-ARIA support in the editor's markup but I'll open a separate issue for that 😄 .

There's a short discussion regarding how to handle keyboard trapping with regards to the text fields: '[What keyboard input makes sense for tabbing out of a rich text field?](https://ux.stackexchange.com/questions/67542/what-keyboard-input-makes-sense-for-tabbing-out-of-a-rich-text-field)', and also a case study from the WCAG '[No Keyboard Trap](https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-trapping.html)'.

I can see two fixes for this issue:
1. Add config property to permanently disable tab key input, or
2. Binding the esc key to temporarily disable the tab key input until the user starts typing again to allow them to hit tab again to leave the text area

**Comment by qkevinto** [comment]
Upon further investigation it seem as though the tab related feature has since somewhat become become a single function:

```
var preventTab = function preventTab(event) {
  if (event.which === 9) event.preventDefault();
};
```

Which gets bound to the content element: `settings.element.content.onkeydown = preventTab;`

So I could simply disable it by doing something like `pellEditor.content.onkeydown = undefined;`. Out of curiosity though, what's the purpose of `preventDefault`ing the tab key and doing nothing?

**Comment by antoninadert** [comment]
Hello, I believe that Tab feature should allow to indent and shift-tab to un-indent the content.

In this case we need a special solution for Accessibility (AX)

**Comment by notlmn** [comment]
Opinion:

What I would like to think is that there is no use of a Tab in content writing. I've never seen anyone use Tab in writings (except in code, which is either copy-pasted or embedded), maybe to indent the start of each paragraph in books.

So I suggest focus should not be trapped, because the default tab behavior cannot be reassigned to other shortcuts as mentioned in stack exchange answer!

**Comment by antoninadert** [comment]
Well it really depends.
In word text editor you can tab
In outlook to write emaim you can tab.

You need another mechanism like echap I believe

**Comment by qkevinto** [comment]
In my opinion, giving people the option to turn off tabbing completely is great. 

But regardless of whether people choose to turn keep it on or off for their implementation, a mechanism like the `esc` key should be available (and possibly  with some help message to let users know they can `esc`) to allow leaving the text field and continue with the document flow.
